src/subsplit_dag.hpp

```diff
@@ -60,6 +60,27 @@ class SubsplitDAG {
     SizeVector new_node_ids, new_edge_idxs, node_reindexer, edge_reindexer;
   };
 
+  // Nearest Neighbor Interchange Operations
+  using NNIOperation = std::pair<Bitset, Bitset>;
```

Comment:
I love that you're defining a type here. 

1. Are you sure that we don't want to make this a full-blown object rather than a `using`? Perhaps we would want methods or to add things later? 
2. If not, perhaps add a little bit of docs about how we encode an NNI as a pair of bitsets?

Reply:
That makes sense.  I'll do that.

Reply:
Great. Having an object will allow you to use descriptive names rather than `first` or `second`... unless somehow those are the best names! 😆

Reply:
Haha... I think we can do little better.

Reply:
Okay, updated NNI's as classes.